### PR TITLE
MINOR: Add more execs to list of supported execs

### PR DESCRIPTION
**Pull request by andygrove**
# Which issue does this PR close?

N/A

 # Rationale for this change

I was reading the docs and we have a list of supported physical plans that is out of date and it did not mention that we support joins.

# What changes are included in this PR?

Added some more execs but not all of them

# Are there any user-facing changes?

No

### CODE REVIEW

File: datafusion/core/src/lib.rs [hunk 1]
```diff
@@ -161,11 +161,17 @@
...
+//! * Sort Merge Join: [`SortExec`](physical_plan::sort_merge_join::SortMergeJoinExec)
+//! * Union: [`SortExec`](physical_plan::union::UnionExec)
 //! * Sort: [`SortExec`](physical_plan::sorts::sort::SortExec)
 //! * Coalesce partitions: [`CoalescePartitionsExec`](physical_plan::coalesce_partitions::CoalescePartitionsExec)
 //! * Limit: [`LocalLimitExec`](physical_plan::limit::LocalLimitExec) and [`GlobalLimitExec`](physical_plan::limit::GlobalLimitExec)
-//! * Scan a CSV: [`CsvExec`](physical_plan::file_format::CsvExec)
-//! * Scan a Parquet: [`ParquetExec`](physical_plan::file_format::ParquetExec)
+//! * Scan CSV: [`CsvExec`](physical_plan::file_format::CsvExec)
+//! * Scan Parquet: [`ParquetExec`](physical_plan::file_format::ParquetExec)
+//! * Scan Avro: [`ParquetExec`](physical_plan::file_format::AvroExec)
```
Comment (alamb) [inline datafusion/core/src/lib.rs:173]:
```suggestion
//! * Scan Avro: [`AvroExec`](physical_plan::file_format::AvroExec)
```

File: datafusion/core/src/lib.rs [hunk 2]
```diff
@@ -161,11 +161,17 @@
 //! * Projection: [`ProjectionExec`](physical_plan::projection::ProjectionExec)
 //! * Filter: [`FilterExec`](physical_plan::filter::FilterExec)
 //! * Grouped and non-grouped aggregations: [`AggregateExec`](physical_plan::aggregates::AggregateExec)
+//! * Hash Join: [`HashJoinExec`](physical_plan::hash_join::HashJoinExec)
+//! * Cross Join: [`HashJoinExec`](physical_plan::cross_join::CrossJoinExec)
+//! * Sort Merge Join: [`SortExec`](physical_plan::sort_merge_join::SortMergeJoinExec)
```
Comment (alamb) [inline datafusion/core/src/lib.rs:166]:
```suggestion
//! * Sort Merge Join: [`SortMergeJoinExec`](physical_plan::sort_merge_join::SortMergeJoinExec)
```

File: datafusion/core/src/lib.rs [hunk 3]
```diff
@@ -161,11 +161,17 @@
...
+//! * Union: [`SortExec`](physical_plan::union::UnionExec)
 //! * Sort: [`SortExec`](physical_plan::sorts::sort::SortExec)
 //! * Coalesce partitions: [`CoalescePartitionsExec`](physical_plan::coalesce_partitions::CoalescePartitionsExec)
 //! * Limit: [`LocalLimitExec`](physical_plan::limit::LocalLimitExec) and [`GlobalLimitExec`](physical_plan::limit::GlobalLimitExec)
-//! * Scan a CSV: [`CsvExec`](physical_plan::file_format::CsvExec)
-//! * Scan a Parquet: [`ParquetExec`](physical_plan::file_format::ParquetExec)
+//! * Scan CSV: [`CsvExec`](physical_plan::file_format::CsvExec)
+//! * Scan Parquet: [`ParquetExec`](physical_plan::file_format::ParquetExec)
+//! * Scan Avro: [`ParquetExec`](physical_plan::file_format::AvroExec)
+//! * Scan newline-delimited JSON: [`ParquetExec`](physical_plan::file_format::NdJsonExec)
```
Comment (alamb) [inline datafusion/core/src/lib.rs:174]:
```suggestion
//! * Scan newline-delimited JSON: [`NdJsonExec`](physical_plan::file_format::NdJsonExec)
```

File: datafusion/core/src/lib.rs [hunk 4]
```diff
@@ -161,11 +161,17 @@
 //! * Projection: [`ProjectionExec`](physical_plan::projection::ProjectionExec)
 //! * Filter: [`FilterExec`](physical_plan::filter::FilterExec)
 //! * Grouped and non-grouped aggregations: [`AggregateExec`](physical_plan::aggregates::AggregateExec)
+//! * Hash Join: [`HashJoinExec`](physical_plan::hash_join::HashJoinExec)
+//! * Cross Join: [`HashJoinExec`](physical_plan::cross_join::CrossJoinExec)
+//! * Sort Merge Join: [`SortExec`](physical_plan::sort_merge_join::SortMergeJoinExec)
+//! * Union: [`SortExec`](physical_plan::union::UnionExec)
```
Comment (alamb) [inline datafusion/core/src/lib.rs:167]:
```suggestion
//! * Union: [`UnionExec`](physical_plan::union::UnionExec)
```

File: datafusion/core/src/lib.rs [hunk 5]
```diff
@@ -161,11 +161,17 @@
 //! * Projection: [`ProjectionExec`](physical_plan::projection::ProjectionExec)
 //! * Filter: [`FilterExec`](physical_plan::filter::FilterExec)
 //! * Grouped and non-grouped aggregations: [`AggregateExec`](physical_plan::aggregates::AggregateExec)
+//! * Hash Join: [`HashJoinExec`](physical_plan::hash_join::HashJoinExec)
+//! * Cross Join: [`HashJoinExec`](physical_plan::cross_join::CrossJoinExec)
```
Comment (alamb) [inline datafusion/core/src/lib.rs:165]:
```suggestion
//! * Cross Join: [`CrossJoinExec`](physical_plan::cross_join::CrossJoinExec)
```

**Review by alamb — APPROVED** [review]

File: datafusion/core/src/lib.rs (hunk 2 above)
Comment (andygrove) [reply datafusion/core/src/lib.rs:166]: Wow, I really messed this up. Sorry for the extra work.

Comment (alamb) [reply datafusion/core/src/lib.rs]: No apologies needed -- Thank you for improving the docs. I am just glad a review offered some value 😆

**Pull request merged**